torchvision/prototype/transforms/_transform.py

```diff
@@ -54,6 +58,41 @@ def extra_repr(self) -> str:
 
         return ", ".join(extra)
 
+    # FIXME explain everything below
+
+    _v1_transform_cls: Optional[Type[nn.Module]] = None
+
+    def _extract_params_for_v1_transform(self) -> Dict[str, Any]:
+        common_attrs = nn.Module().__dict__.keys()
+        params = {
+            attr: value
+            for attr, value in self.__dict__.items()
+            if not attr.startswith("_") and attr not in common_attrs
+        }
+
+        if "fill" not in params:
+            return params
+
+        fill_type_dict = params.pop("fill")
+        fill_candidates = [fill_type_dict[typ] for typ in [torch.Tensor, datapoints.Image]]
+        try:
+            fill = next(filter(lambda candidate: candidate is not None, fill_candidates))
+        except StopIteration:
+            fill = None
+        params["fill"] = fill
+        return params
+
+    def __init_subclass__(cls) -> None:
+        if cls._v1_transform_cls is not None and hasattr(cls._v1_transform_cls, "get_params"):
+            cls.get_params = cls._v1_transform_cls.get_params  # type: ignore[attr-defined]
```

Comment:
IIUC this is for exposing the `get_params()` methods and not immediately related to torchscript support (or is it?). If not, should we save it for another PR?

Reply:
Yeah, earlier versions of this patch tied the two together. But now I can cleanly put this into a separate PR. Will do.